### PR TITLE
fix(react-grid): calculate summary for collapsed groups under first level

### PR DESCRIPTION
**Pull request by LazyLahtak**

### CODE REVIEW

File: packages/dx-grid-core/src/plugins/integrated-summary/computeds.ts [hunk 1]
```diff
@@ -97,6 +94,17 @@ export const groupSummaryValues: GroupSummaryValuesFn = (
         row,
         rows: [],
       });
+      levelIndex = levels.findIndex(level => level.levelKey === levelKey);
```
Comment (ushkal) [inline packages/dx-grid-core/src/plugins/integrated-summary/computeds.ts]: A function should be extracted

File: packages/dx-grid-core/src/plugins/integrated-summary/computeds.ts [hunk 2]
```diff
@@ -97,6 +97,17 @@ export const groupSummaryValues: GroupSummaryValuesFn = (
         row,
         rows: [],
       });
+      levelIndex = getLevelIndex(levelKey);
+    }
+    const isCollapsedNestedGroupRow = collapsedRows && levelIndex > 0;
+    if (!levelKey || isCollapsedNestedGroupRow) {
+      levels.forEach((level) => {
+        if (!levelKey) {
+          level.rows.push(row);
```
Comment (DmitryBogomolov) [inline packages/dx-grid-core/src/plugins/integrated-summary/computeds.ts]: Why is array modified in place here and recreated in the `else` case?

```js
level.rows.push(...collapsedRows);
```

File: packages/dx-grid-core/src/plugins/integrated-summary/computeds.ts [hunk 3]
```diff
@@ -97,6 +97,17 @@ export const groupSummaryValues: GroupSummaryValuesFn = (
         row,
         rows: [],
       });
+      levelIndex = getLevelIndex(levelKey);
+    }
+    const isCollapsedNestedGroupRow = collapsedRows && levelIndex > 0;
+    if (!levelKey || isCollapsedNestedGroupRow) {
+      levels.forEach((level) => {
+        if (!levelKey) {
```
Comment (DmitryBogomolov) [inline packages/dx-grid-core/src/plugins/integrated-summary/computeds.ts]: The condition is checked on each iteration. Code can be replaced with something like this:
```js
const rowsToAppend = !levelKey ? [row] : collapsedRows;
levels.forEach((level) => {
  level.rows.push(...rowsToAppend);
});
```

File: packages/dx-grid-core/src/plugins/integrated-summary/computeds.ts [hunk 4]
```diff
@@ -71,20 +71,20 @@ export const groupSummaryValues: GroupSummaryValuesFn = (
   getCellValue,
   getRowLevelKey,
   isGroupRow,
+  getCollapsedRows,
   calculator = defaultSummaryCalculator,
 ) => {
   let levels: any[] = [];
```
Comment (DmitryBogomolov) [inline packages/dx-grid-core/src/plugins/integrated-summary/computeds.ts]: Isn't there `Level` type?

File: packages/dx-grid-core/src/plugins/integrated-summary/computeds.ts (hunk 2 above)
Comment (ushkal) [reply packages/dx-grid-core/src/plugins/integrated-summary/computeds.ts]: Agreed, that would be more consistent

**Pull request merged**